js/packages/quary-extension-bus/src/result.ts

```diff
@@ -70,7 +50,91 @@ export function collectResults<T, E>(
     }
   }
   if (errors.length > 0) {
-    return Err(new Error(`Errors: ${errors.join(', ')}`))
+    return Err(errors)
   }
   return Ok(values)
 }
+
+export enum ErrorCodes {
+  OK = 0,
+  CANCELLED = 1,
+  UNKNOWN = 2,
+  INVALID_ARGUMENT = 3,
+  DEADLINE_EXCEEDED = 4,
+  NOT_FOUND = 5,
+  ALREADY_EXISTS = 6,
+  PERMISSION_DENIED = 7,
+  RESOURCE_EXHAUSTED = 8,
+  FAILED_PRECONDITION = 9,
+  ABORTED = 10,
+  OUT_OF_RANGE = 11,
+  UNIMPLEMENTED = 12,
+  INTERNAL = 13,
+  UNAVAILABLE = 14,
+  DATA_LOSS = 15,
+  UNAUTHENTICATED = 16,
+}
+
+/**
+ * Converts an error code to a human-readable string.
+ *
+ */
+export function codeToString(code: ErrorCodes): string {
+  switch (code) {
+    case ErrorCodes.OK:
+      return 'Ok'
+    case ErrorCodes.CANCELLED:
+      return 'Cancelled'
+    case ErrorCodes.UNKNOWN:
+      return 'Unknown Error'
+    case ErrorCodes.INVALID_ARGUMENT:
+      return 'Invalid Argument'
+    case ErrorCodes.DEADLINE_EXCEEDED:
+      return 'Deadline Exceeded'
+    case ErrorCodes.NOT_FOUND:
+      return 'Not Found'
+    case ErrorCodes.ALREADY_EXISTS:
+      return 'Already Exists'
+    case ErrorCodes.PERMISSION_DENIED:
+      return 'Permission Denied'
+    case ErrorCodes.RESOURCE_EXHAUSTED:
+      return 'Resource Exhausted'
+    case ErrorCodes.FAILED_PRECONDITION:
+      return 'Failed Precondition'
+    case ErrorCodes.ABORTED:
+      return 'Aborted'
+    case ErrorCodes.OUT_OF_RANGE:
+      return 'Out of Range'
+    case ErrorCodes.UNIMPLEMENTED:
+      return 'Unimplemented'
+    case ErrorCodes.INTERNAL:
+      return 'Internal'
+    case ErrorCodes.UNAVAILABLE:
+      return 'Unavailable'
+    case ErrorCodes.DATA_LOSS:
+      return 'Data Loss'
+    case ErrorCodes.UNAUTHENTICATED:
+      return 'Unauthenticated'
+    default:
+      throw new Error(`Unknown error code: ${code} that should never happen`)
+  }
+}
+
+export type QuaryError = {
+  code: ErrorCodes
+  message: string
+  tree?: QuaryError
+}
+
+export function isQuaryError(value: any): value is QuaryError {
```

Comment:
Unknown potentially a more accurate type 
```suggestion
export function isQuaryError(value: unknown): value is QuaryError {
```